### DISCUSSION TITLE
Text SCTRL to use whatever localcoord the player is using.

### DISCUSSION THREAD

**Discussion by Orochikyocr** [ikemen-engine/Ikemen-GO#1125] [Feature Request] [answered]
Text SCTRL uses const240p to calculate its position.
Other sctrls like explod, lifebaraction, etc use the local aspect ratio set by the user.

I develop modules for many purposes and I like to show relevant info to the player. But when going through different aspect ratios the position of the text is affected severely.

I was told to use negative values for this but to be honest, this sound like kinda "tweaky" to be constantly creating different conditions for whatever resolution or aspect ratio people will use for their Ikemen Go installation.
Using helpers is like a big "no", because I develop mostly tag stuff and that means one helper for every char. That is not optimal for both performance and debugging purposes.

I propose to make the SCTRL use the LocalCoord of the screen pack or the localcoord calculated from the resolution chosen by the player.
For those using lifebaraction or any other SCTRL it will be easier to calculate their text{} SCTRL position. 
We can use other triggers like "screenwidth" to calculate the text on the right of the corner. 
This will standardize all your new and old sctrls using the same coordinates, instead of having some using const240p while others use the localcoord. This is confusing for developers.
It will allow to simplify the code and allow the text{} to find the intended position no matter the resolution chosen by the player.

Have a good day.

**Comment by LasombraDemon** [comment]
PotS' idea of defining the LocalCoord as a parameter sounds like the smartest option. We can then use whatever value we want instead of a hardcoded one.

**Reply by Orochikyocr** [reply]
> PotS' idea of defining the LocalCoord as a parameter sounds like the smartest option. We can then use whatever value we want instead of a hardcoded one.

That means setting conditionals for two or more lines that do the same job. it is not the idea.
The idea is, like every other SCTRL, text using either the LOCALCOORD set in the fight.def or the actual screen resolution the player set in their config.
The idea is to simplify code, not to create lines for every single aspect ratio possible.

**Comment by iamno1one** [comment]
i agree

**Comment by potsmugen** [comment]
The idea that Text should use the character's localcoord has merit, I think. It's how everything else works (explods, etc). Someone who's making a hires character shouldn't have to change their points of reference for one sctrl.
It won't solve this issue but still.

EDIT: The issue might also be that it looks like Text is supposed to work like PosType = Left but it actually doesn't do that.

**Comment by K4thos** [comment, marked as answer]
https://github.com/ikemen-engine/Ikemen-GO/pull/1179
